### PR TITLE
Docs: Move docs from eslint.github.io (fixes #5964)

### LINKED ISSUE #5964: Move all docs into eslint repo
Right now we have a very weird mix of some docs that live in `eslint.github.io` and some that live in `eslint`. I find this pretty confusing (though it's highly likely I'm the one to blame for this!). For instance, there are files here:

https://github.com/eslint/eslint.github.io/tree/master/docs/user-guide

That are not here:

https://github.com/eslint/eslint/tree/master/docs/user-guide

I'd like to propose that everything in the `docs` directory on `eslint.github.io` be maintained in the `eslint` repo from now on. That will help us ensure that all documentation changes can happen in one place without any arbitrary distinction (which we appear to have right now).

This would still allow us to update things like the demo, homepage, blog, etc., directly in `eslint.github.io`. I'm talking only about moving those files that are currently in `docs`.

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

**Comment by nzakas** [comment]
Man, our markdown linting rules are rough - that's why the builds are failing. I'll try to fix them up.

File: Makefile.js [hunk 1]
```diff
@@ -521,12 +522,10 @@ target.gensite = function(prereleaseVersion) {
 
     var docFiles = [
         "/rules/",
-        "/user-guide/command-line-interface.md",
-        "/user-guide/configuring.md",
-        "/developer-guide/code-path-analysis.md",
-        "/developer-guide/nodejs-api.md",
-        "/developer-guide/working-with-plugins.md",
-        "/developer-guide/working-with-rules.md"
+        "/user-guide/",
```
Comment (pedrottimark) [inline Makefile.js]: @nzakas Now that all the `docFiles` are directory paths, `if (test("-f", fullPath)) { ... }` never executes in step 2 below.

Can you confirm whether that logic has become obsolete, or needs to be rewritten?

If obsolete, I can just delete it when I submit pull request to automatically add rule meta data to docs.

Comment (nzakas) [inline Makefile.js]: `find()` returns an array of filenames, so that condition is still triggered.

**Comment by Arcanemagus** [comment]
Looks like the `___.html` links in the "index" `README.md` files weren't corrected to `____.md`, should I file a new issue on that, review the necessary changes here, or is this comment enough?

**Pull request merged**